src/menu.c

```diff
@@ -12,17 +11,16 @@
  * @param menu_text The text presented at the top of the menu
  * @param help_text The text printed when the user asks for help
  */
-#define display_menu(options, menu_text, help_text) _print_menu(options, sizeof(options)/sizeof(char*), menu_text, help_text)
+#define display_menu(options, menu_text, help_text) _display_menu(options, sizeof(options)/sizeof(char*), menu_text, help_text)
 
 int _display_menu(char **options, int options_len, char *menu_text, char *help_text) {
-    // TODO: figure out how to reprint the menu
-    printf("%s\n", menu_text);
-    printf("Enter !q to quit, or !h to get help");
-    // Print the menu options
-    for (int i = 0; i < options_len; i++) {
-        printf("[%d] %s\n", i+1, options[i]);
-    }
     while (1) {
+        printf("%s\n", menu_text);
+        printf("Enter !q to quit, or !h to get help\n");
+        // Print the menu options
+        for (int i = 0; i < options_len; i++) {
+            printf("[%d] %s\n", i+1, options[i]);
+        }
         printf("> ");
         // Let's hope that no one decides to enter more than 16 characters in the menu :^)
         // We need to read the input into a string first, because we need to check whether it's a number, a command, or something invalid
```

Comment:
Avoid stuff like this, were we make assumptions which could potentially crash the program. As you are just reading a string, an alternative could be fgets(input, 16, stdin). This programmatically limits the input length, throwing away anything longer than 16 chars